### ISSUE TITLE
Generate a emtpy TestcontainersConfiguration for Couchbase 

### ISSUE THREAD

**Issue by hantsy** [spring-io/initializr#1594]
I tried to generate a Spring boot with following config in the start.spring.io.
* Maven
* java 21
* Dependencies: Reactive Web, Spring Data Couchbase Reactive, Testcontainers

In the generated files, the `TestcontainersConfiguration` is an empty class.

I tried to add a bean declaration as usual. 

```java
@Configuration(proxyBeanMethods = false)
public class TestcontainersConfiguration {
    private static final String COUCHBASE_IMAGE_NAME = "couchbase";
    private static final String DEFAULT_IMAGE_NAME = "couchbase/server:7";
    private static final DockerImageName DEFAULT_IMAGE = DockerImageName.parse(COUCHBASE_IMAGE_NAME)
            .asCompatibleSubstituteFor(DEFAULT_IMAGE_NAME);

    @Bean
    @ServiceConnection
    public CouchbaseContainer couchbaseContainer() {
        return new CouchbaseContainer(DEFAULT_IMAGE)
                .withCredentials("Administrator", "password")
                .withBucket(new BucketDefinition("demo").withPrimaryIndex(true))
                .withStartupAttempts(5)
                .withStartupTimeout(Duration.ofSeconds(120))
                .waitingFor(Wait.forHealthcheck());
    }
}
```

And in the Repository tests, import it, but it report can not find the `Repository` bean. Check the [PostRepositoryTests](https://github.com/hantsy/spring-reactive-sample/blob/master/boot-data-couchbase/src/test/java/com/example/demo/PostRepositoryTest.java).

There is an alternative [PostRepositoryWithTestcontainersTest](https://github.com/hantsy/spring-reactive-sample/blob/master/boot-data-couchbase/src/test/java/com/example/demo/PostRepositoryWithTestcontainersTest.java) that uses traditional `@Testcontainers` and `DynamicPropertyRegistry`, the test is passed.

**Comment by mhalbritter** [comment]
I guess you're asking for code generation support for the `CouchbaseContainer`?

**Comment by hantsy** [comment]
Yeah, for other dbs it generates the testcontainerConfig with a container bean correctly. 

A strange thing about couchbase is when I added the `CouchbaseServer` bean manually, I found it depends on `spring.data.couchbase.bucketName` property. Without this property, the tests will be failed with a `Repository not found` when running [the repository tests with a `@DataCouchbaseTest`](https://github.com/hantsy/spring-reactive-sample/blob/master/boot-data-couchbase/src/test/java/com/example/demo/PostRepositoryTest.java).
